sql/analyzer/symbol_resolution.go

```diff
@@ -78,7 +78,15 @@ func pruneTables(ctx *sql.Context, a *Analyzer, n sql.Node, s *Scope, sel RuleSe
 		switch n := n.(type) {
 		case *plan.ResolvedTable:
 			return pruneTableCols(n, parentCols, parentStars, unqualifiedStar)
-		case plan.JoinNode, *plan.CrossJoin, *plan.IndexedJoin, *plan.Filter,
+		case plan.JoinNode:
```

Comment:
perhaps too broad, it'd be nice to include the expression from the JSONTable and still prune the left side